### ISSUE TITLE
Set up encrypted credentials in Travis to enable push to DockerHub

### ISSUE THREAD

**Issue by georgemccabe** [dtcenter/METplus#634]
## Describe the Task ##
Store secret credentials in Travis so that we can log into DockerHub and push images through Travis without authenticating by hand.

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
None

### Funding Source ###
None

### Assignee ###
- [X] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [X] Select **component(s)**
- [X] Select **priority**
- [X] Select **requestor(s)**

### Projects and Milestone ###
- [X] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [X] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
If other projects will utilize Travis and Docker together, document the process to set this up so that the other projects can also benefit.

- [] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [x] Add/update log messages for easier debugging.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [x] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by georgemccabe** [comment]
Sub-task of #633

**Comment by fentoad72** [comment]
What was the update from SNAT about group/role accounts?

**Comment by georgemccabe** [comment]
I am still waiting to hear back from SNAT about the group accounts. In the meantime, I stored my Docker credentials in Travis as encrypted environment variables to make sure the functionality works as intended. Once we get approval to create a group account, we can replace the values with the new user's credentials and everything should work the same.

**Comment by fentoad72** [comment]
I'm looking at pass[https://www.passwordstore.org/(https://www.passwordstore.org/) , which is a unix-based password manager so can be used on Apple, Windows, and linux, something that is more secure than transmitting a cleartext password.

**Comment by georgemccabe** [comment]
We haven't heard back from SNAT about group accounts and Todd has been analyzing the time for each Travis job (#649), so I am moving this to beta2.

**Comment by fentoad72** [comment]
I created a branch feature_634_credentials and did some tests.

1) There is no .docker directory, and no .docker/config.json on the Travis Virtual Machine in either the home directory (/home/travis), the build directory (/home/travis/build/fentoad72), or the METplus directory (/home/travis/build/fentoad72/METplus)
2) In the job log, DOCKER_USERNAME=[secure] and DOCKER_PASSWORD=[secure].  Even the name of the build directory is listed as /home/travis/build/[secure]
3) I ran "env" and DOCKER_USERNAME and DOCKER_PASSWORD are [secure]
4) I ran "docker swarm init" and "docker secret ls" and neither of DOCKER USER/PASS is listed
5) I ran "cat .bashrc" and the .bashrc on /home/travis is not the same as on my local machine (apostrophe laptop)
6)  It appears that the Travis VM does not copy the .* files from my local home directory; in fact, the /home/travis directory has its own set of .* files.  Meanwhile, /home/travis/build/fentoad72 is empty except for METplus.

Since the password store directory (.password-store) is a hidden file in my home directory, it is likely it won't be copied; so pass is unlikely to work inside of Travis.  As George pointed out, Travis keeps the Docker credentials secret ([secure]) and my attempts to print them out all failed.

I'm inclined to close this ticket as it appears Travis keeps the Docker credentials secure and there is no need for a 3rd-party app

Here's the output from my testing:

[travis634_lastlog.txt](https://github.com/dtcenter/METplus/files/5539125/travis634_lastlog.txt)
